src/main/scala/org/apache/mesos/chronos/scheduler/jobs/constraints/LikeConstraint.scala

```diff
@@ -0,0 +1,30 @@
+package org.apache.mesos.chronos.scheduler.jobs.constraints
+
+import org.apache.mesos.Protos
+import scala.collection.JavaConversions._
+
+case class LikeConstraint(attribute: String, value: String) extends Constraint {
+
+  val regex = value.r
+
+  def matches(attributes: Seq[Protos.Attribute]): Boolean = {
+    attributes.exists { a =>
+      if (a.getName == attribute) {
+        a.getType match {
+          case Protos.Value.Type.SCALAR =>
+            return regex.pattern.matcher(a.getScalar.getValue.toString).matches()
```

Comment:
Please don't use return - this is only a short circuit that is not needed here, since exist will return directly after an element is found.